### PR TITLE
fix: conda environments

### PR DESCRIPTION
**Pull request by znorgaard**
Also includes #93 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/fastquorum/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/fastquorum _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/bwa/index/environment.yml
Comment (SPPearce) [inline modules/nf-core/bwa/index/environment.yml:1]: Why have these gained the defaults channel back?

File: CHANGELOG.md
Comment (SPPearce) [inline CHANGELOG.md:1]: I think we want to include a table saying explicitly what versions have changed.

**Comment by SPPearce** [comment]
At some point we should move to the nf-core modules repo versions of the fgbio modules that are in there (i.e. the ones that aren't a combination of two subtools. 
Currently we have one nf-core module from fgbio, `fgbio sortbam`. If we want to update the version of that like you have here, it is good practice to then use a patch diff (`nf-core modules diff`) to ensure that things stay in sync as we update the nf-core modules. It'd be good generally to update all the nf-core modules to 2.4.0 on the modules repo, then that wouldn't be needed.

Not sure why the `defaults` channels (and the `name:` field) have turned back up in the nf-core modules, what did you do to update those? There is a technical reason to remove the `defaults` (something to do with anaconda requiring a license, I can dig out the information if you want).

File: modules/nf-core/bwa/index/environment.yml
Comment (znorgaard) [reply modules/nf-core/bwa/index/environment.yml:1]: We can take them out, I just like including it to be explicit. I do see it looks like an intentional change in the template.

Comment (SPPearce) [reply modules/nf-core/bwa/index/environment.yml:1]: Yes, we don't want to use anything from the `defaults` channel.
https://www.datacamp.com/blog/navigating-anaconda-licensing
> Companies with 200 or more employees must purchase a paid license to access the default channel or use Anaconda Distribution.

**Review by SPPearce — APPROVED** [review]
I've just added patch diff files by running `nf-core modules patch` on each module - these will ensure that the changes to the environment will remain if we update the modules from nf-core.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+  | Module         | Previous SHA | New SHA |
+  | -------------- | ------------ | ------- |
+  | bwa/index      | e0ff65e      | 6666521 |
+  | fastqc         | b49b899      | 6666521 |
+  | fgbio/sortbam  | 2fc7438      | bc6d86f |
+  | multiqc        | fe9614c      | cf17ca4 |
+  | samtools/dict  | 3c8fd07      | b13f07b |
+  | samtools/faidx | 04fbbc7      | b13f07b |
+  | samtools/merge | 04fbbc7      | b13f07b |
```
[the remark this answers is not in the document]
Comment (znorgaard) [reply CHANGELOG.md:26]: `nf-core modules list local`

**Comment by znorgaard** [comment]
Something about the stacked PR merge is making linting break. Going to revert that merge to see if it will fix the problem. Shouldn't impact anything else here.

**Pull request merged**